src/rapidhtml/callbacks.py

```diff
@@ -0,0 +1,243 @@
+            "beforeend",
+            "afterend",
+            "delete",
+            "none",
+        ]
+    ] = "innerHTML"
+    swap_oob: Optional[
+        Literal[
+            "innerHTML",
+            "outerHTML",
+            "textContent",
+            "beforebegin",
+            "afterbegin",
+            "beforeend",
+            "afterend",
+            "delete",
+            "none",
+        ]
+    ] = None
+    target: Optional[str] = None
+    trigger: Optional[str] = None
+    vals: Optional[dict] = None
+    boost: Optional[bool] = None
+    confirm: Optional[str] = None
+    disable: Optional[bool] = None
+    disabled_elt: Optional[str] = None
+    disinherit: Optional[str] = None
+    encoding: Optional[str] = None
+    ext: Optional[str] = None
+    headers: Optional[dict] = None
+    history: Optional[bool] = None
+    history_elt: Optional[bool] = None
+    include: Optional[str] = None
+    indicator: Optional[str] = None
+    inherit: Optional[str] = None
+    params: Optional[str] = None
+    preserve: Optional[bool] = None
+    prompt: Optional[str] = None
+    replace_url: Optional[str] = None
+    request: Optional[str | dict] = None
+    sync: Optional[
+        Literal[
+            "drop",
+            "abort",
+            "replace",
+            "queue",
+            "queue first",
+            "queue last",
+            "queue all",
+        ]
+    ] = None
+    validate: Optional[bool] = None
+
+    def get_data(self) -> tuple[Callable, str, dict]:
+        """Get the data of the callback.
+
+        Returns:
+            tuple[Callable, str, dict]: The function, method, and attributes of
+            the callback.
+        """
+        attrs = {}
+        if self.on:
+            event = self.on[0]
+            attrs[f"hx_on:{event}"] = self.on[1]
+        if self.push_url:
+            attrs["hx_push_url"] = self.push_url
+        if self.select:
+            attrs["hx_select"] = self.select
+        if self.select_oob:
+            attrs["hx_select_oob"] = self.select_oob
+        if self.swap:
+            attrs["hx_swap"] = self.swap
+        if self.swap_oob:
+            attrs["hx_swap_oob"] = self.swap_oob
+        if self.target:
+            attrs["hx_target"] = self.target
+        if self.trigger:
+            attrs["hx_trigger"] = self.trigger
+        if self.vals:
+            attrs["hx_vals"] = json.dump(self.vals)
+        if self.boost:
+            attrs["hx_boost"] = "true" if self.boost else "false"
+        if self.confirm:
+            attrs["hx_confirm"] = self.confirm
+        if self.disable:
+            attrs["hx_disable"] = self.disable
+        if self.disabled_elt:
+            attrs["hx_disabled_elt"] = self.disabled_elt
+        if self.disinherit:
+            attrs["hx_disinherit"] = self.disinherit
+        if self.encoding:
+            attrs["hx_encoding"] = self.encoding
+        if self.ext:
+            attrs["hx_ext"] = self.ext
+        if self.headers:
```

Comment:
I feel like these ifs should be:

```python
if {param} is not None:
```

Because if the user passes a value, but the value is false-y, currently it'd not get added to the dict